engine/configurator/conf_yaml.c

```diff
@@ -153,167 +235,232 @@ parse_logic_expr_str(const char *str, te_bool *res, te_kvpair_h *expand_vars)
-                      (const char *)k->data.scalar.value);
-                return TE_EINVAL;
+                case CS_YAML_OBJECT_D:
+                    rc = parse_config_str(v, &obj->d);
+                    break;
+
+                case CS_YAML_OBJECT_OID:
+                    rc = parse_config_str(v, &obj->oid);
+                    break;
+
+                case CS_YAML_OBJECT_ACCESS:
+                    rc = parse_config_str_by_mapping(v, &obj->access,
+                                        cs_yaml_object_access_fields_mapping);
+                    break;
+
+                case CS_YAML_OBJECT_TYPE:
+                    rc = parse_config_str_by_mapping(v, &obj->type,
+                                                     cfg_cvt_mapping);
+                    if (rc ==0 && obj->type == CVT_UNSPECIFIED)
+                        rc = TE_EINVAL;
+                    break;
+
+                case CS_YAML_OBJECT_UNIT:
+                    rc = parse_config_str_by_mapping(v, &temp,
+                                                     cs_yaml_bool_mapping);
+                    obj->unit = temp;
+                    break;
+
+                case CS_YAML_OBJECT_DEF_VAL:
+                    rc = parse_config_str(v, &obj->def_val);
+                    break;
+
+                case CS_YAML_OBJECT_VOLAT:
+                    rc = parse_config_str_by_mapping(v, &temp,
+                                                     cs_yaml_bool_mapping);
+                    obj->volat = temp;
+                    break;
+
+                case CS_YAML_OBJECT_SUBSTITUTION:
+                    rc = parse_config_str_by_mapping(v, &temp,
+                                                     cs_yaml_bool_mapping);
+                    obj->substitution = temp;
+                    break;
+
+                case CS_YAML_OBJECT_NO_PARENT_DEP:
+                    rc = parse_config_str_by_mapping(v, &temp,
+                                          cs_yaml_object_no_parent_dep_mapping);
+                    obj->type = temp;
+                    break;
+
+                case CS_YAML_OBJECT_DEPENDS:
+                    rc = parse_config_yaml_dependency(ctx->doc, v, obj);
+                    break;
+
+                default:
+                    ERROR(CS_YAML_ERR_PREFIX "failed to recognise the command '%s'",
+                          (const char *)k->data.scalar.value);
+                    rc = TE_EINVAL;
             }
-            break;
+            if (rc != 0)
+            {
+                ERROR(CS_YAML_ERR_PREFIX "failed to process %s "
+                      "attribute at " YAML_NODE_LINE_COLUMN_FMT "",
+                      (const char *)k->data.scalar.value,
+                      YAML_NODE_LINE_COLUMN(k));
+                return rc;
+            }
+        } while (++pair < n->data.mapping.pairs.top);
+    }
+    else
+    {
+        ERROR(CS_YAML_ERR_PREFIX
+              "found the instance node to be not a mapping node");
```

Comment:
Applies everywhere: "found X to be not" is just a pile of junk words, write just "X is not", it's shorter and way easy to grasp.

Reply:
OK, I've made it everywhere.